data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retrieving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
+
+mod asset {
+    //! This module contains `AssetEvent`, `AssetDefinitionEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
```

Comment:
how come we're allowing missing docs?

Reply:
I'm who added this attribute to event enums.
The intention was that the typenames should be enough descriptive for subscribers

Reply:
I think some docs could be useful here. Not to explain what the field is, but when to use it. This is a library after all.
For example: 
```/// Created. The `Entity` has appeared on the blockchain for the first time. Do not use for `Mint` and `Grant` instructions. ```

Reply:
They would not have to be so detailed
as the event status is now more specific than the previous where one has been paired with various entities

Reply:
So we will do it later. I'm closing this conversation